File: source/orch_mgr/java/src/com/formationds/xdi/local/Domain.java
package com.formationds.xdi.local;
/*
 * Copyright 2014 Formation Data Systems, Inc.
 */

import org.hibernate.Hibernate;

import javax.persistence.*;

@Entity
@Table(name="domain")
public class Domain implements Persistent {
    private long id;
    private String name;

    public Domain() {
    }

    public Domain(String name) {
        this.name = name;
    }

    @Id
    @GeneratedValue
    @Column(name="id")
    public long getId() {
        return id;
    }

    @Override
    public void initialize() {
        Hibernate.initialize(this);
    }

    @Column(name="name", nullable = false, unique = true)
    public String getName() {
        return name;
    }

    public void setId(long id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: source/orch_mgr/om-domain/domain.cpp
/*
 * Copyright 2014 by Formation Data Systems, Inc.
 */
#include <om-domain.h>

namespace fds {

OM_NodeWrkFlow::~OM_NodeWrkFlow() {}
OM_NodeWrkFlow::OM_NodeWrkFlow(char const *const name) : Module(name)
{
}

int
OM_NodeWrkFlow::mod_init(SysParams const *const param)
{
    Module::mod_init(param);
    return 0;
}

void
OM_NodeWrkFlow::mod_startup()
{
}

void
OM_NodeWrkFlow::mod_enable_service()
{
}

void
OM_NodeWrkFlow::mod_shutdown()
{
}

}  // namespace fds
